src/bp-members/classes/class-bp-members-notices-admin.php

```diff
@@ -0,0 +1,405 @@
+ *
+ * @package buddypress\bp-members\classes\class-bp-members-notices-admin
+ * @since 15.0.0
+ */
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * BuddyPress Notices Admin class.
+ */
+#[AllowDynamicProperties]
+class BP_Members_Notices_Admin {
+
+	/**
+	 * The ID returned by `add_users_page()`.
+	 *
+	 * @since 15.0.0
+	 * @var string
+	 */
+	public $screen_id = '';
+
+	/**
+	 * The URL of the admin screen.
+	 *
+	 * @since 15.0.0
+	 * @var string
+	 */
+	public $url = '';
+
+	/**
+	 * The current instance of the BP_Members_Notices_List_Table class.
+	 *
+	 * @since 15.0.0
+	 * @var BP_Members_Notices_List_Table|string
+	 */
+	public $list_table = '';
+
+	/**
+	 * Create a new instance or access the current instance of this class.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @return BP_Members_Notices_Admin
+	 */
+	public static function register_notices_admin() {
+
+		if ( ! is_admin() || ! bp_current_user_can( 'bp_moderate' ) ) {
+			return;
+		}
+
+		$bp = buddypress();
+
+		if ( empty( $bp->members->admin->notices ) ) {
+			$bp->members->admin->notices = new self;
+		}
+
+		return $bp->members->admin->notices;
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 15.0.0
+	 */
+	public function __construct() {
+		$this->setup_globals();
+		$this->setup_actions();
+	}
+
+	/**
+	 * Populate the classs variables.
+	 *
+	 * @since 15.0.0
+	 */
+	protected function setup_globals() {
+		$this->url = add_query_arg( array( 'page' => 'bp-notices' ), bp_get_admin_url( 'users.php' ) );
+	}
+
+	/**
+	 * Add action hooks.
+	 *
+	 * @since 15.0.0
+	 */
+	protected function setup_actions() {
+		add_action( bp_core_admin_hook(), array( $this, 'admin_menu' ) );
+	}
+
+	/**
+	 * Add the 'Site Notices' admin menu item.
+	 *
+	 * @since 15.0.0
+	 */
+	public function admin_menu() {
+		// Bail if current user cannot moderate community.
+		if ( ! bp_current_user_can( 'bp_moderate' ) ) {
+			return false;
+		}
+
+		$this->screen_id = add_users_page(
+			_x( 'Member Notices', 'Notices admin page title', 'buddypress' ),
+			_x( 'Member Notices', 'Admin Users menu', 'buddypress' ),
+			'manage_options',
+			'bp-notices',
+			array( $this, 'admin_index' )
+		);
+
+		add_action( 'load-' . $this->screen_id, array( $this, 'admin_load' ) );
+	}
+
+	/**
+	 * Catch save/update requests or load the screen.
+	 *
+	 * @since 15.0.0
+	 */
+	public function admin_load() {
+		$redirect_to = false;
+
+		if ( ! bp_current_user_can( 'bp_moderate' ) ) {
+			wp_die( esc_html__( 'You are not allowed to manage Member Notices.', 'buddypress' ) );
+		}
+
+		// Catch new notice saves.
+		if ( ! empty( $_POST['bp_notice']['send'] ) || ! empty( $_POST['bp_notice']['update'] ) ) {
+
+			check_admin_referer( 'save-notice', 'ns-nonce' );
+
+			$notice = bp_parse_args(
+				$_POST['bp_notice'],
+				array(
+					'id'       => 0,
+					'title'    => '',
+					'content'  => '',
+					'target'   => '',
+					'priority' => 2,
+					'url'     => '',
+					'text'     => '',
+				)
+			);
```

Comment:
While testing this, and looking at the code, it seems title and content should be required. But there is no _visual_ indication of that. Try to send a notice without any content and you get a successful message.

![CleanShot 2024-09-02 at 13 14 08@2x](https://github.com/user-attachments/assets/99941f88-c6a3-4828-b7bb-84eeab76e825)

Reply:
Good catch! I'll improve this.

Reply:
<img width="1245" alt="user-feedback" src="https://github.com/user-attachments/assets/6205e338-a407-483f-aa97-8eb7a2f44d4c">

When title and/or content are not provided it now displays a more meaningful error + highlights the required fields like WordPress does for categories/tags.